### PR TITLE
[feat] Allow self loops in module mappings

### LINKED ISSUE #843: Allow self referring module mappings
Running ReFrame with the following will issue an error of cyclic dependency:

```
-M 'A: A B'
```
This scenario is useful in testing when you want to load an additional module without touching the test, and actually it is not a cyclic dependency.

@lucamar has also came across this problem in the past if I'm not mistaken.

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #843

### CODE REVIEW

File: reframe/core/modules.py [hunk 1]
```diff
@@ -109,13 +109,19 @@ def resolve_module(self, name):
...
+        loop_nodes = set()
         while unvisited:
             node, parent = unvisited.pop()
-
             # Adjust the path
             while path and path[-1] != parent:
                 path.pop()
 
+            # Handle modules mappings with self loops
+            if node == parent and node not in loop_nodes:
```
Comment (vkarak) [inline reframe/core/modules.py]: Why do you need the `loop_nodes` here?

Comment (teojgo) [reply reframe/core/modules.py]: I need it to avoid duplicate modules.

Comment (vkarak) [reply reframe/core/modules.py]: Can't you just use an `OrderedSet` for the `ret` value and convert it to a list at return?

File: reframe/core/modules.py [hunk 2]
```diff
@@ -105,21 +106,19 @@ def resolve_module(self, name):
...
             node, parent = unvisited.pop()
             # Adjust the path
             while path and path[-1] != parent:
                 path.pop()
 
             # Handle modules mappings with self loops
-            if node == parent and node not in loop_nodes:
-                loop_nodes.add(node)
-                ret.append(node)
+            if node == parent and node not in ret:
```
Comment (vkarak) [inline reframe/core/modules.py]: You don't need the `not in ret` part. It's handled the `OrderedSet`.

Comment (teojgo) [reply reframe/core/modules.py]: A ok, I thought that it would update the order if it was already in.

**Pull request merged**